RELEASE.md

```diff
@@ -0,0 +1,120 @@
+# OpenTelemetry Release Processes Overview
+
+## Common Release Practices
+  * **Semantic Versioning (SemVer)** is used in every repository (`vMAJOR.MINOR.PATCH`). See our [Versioning and Stability](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md) policy for more information.
+  * CHANGELOGs keep an **“Unreleased”** section that is promoted to a versioned section during a release.
+  * Releases are tagged from the default branch or a dedicated *release/x.y* branch; tags are always prefixed with **v**.
+  * GitHub Actions prepare release PRs, bump versions, draft release notes, publish artifacts, and create Git tags.
```

Comment:
In OTel Go we do it semi-automated way.
We do not use GitHub Actions for releasing. 
Doesn't it allow any approver to create a release?
Shouldn't only maintainers be allowed to created releases?